server/src/test/java/org/cloudfoundry/identity/uaa/scim/endpoints/UserIdConversionEndpointsTests.java

```diff
@@ -13,65 +13,89 @@
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
 import org.cloudfoundry.identity.uaa.resources.SearchResults;
+import org.cloudfoundry.identity.uaa.scim.ScimUser;
+import org.cloudfoundry.identity.uaa.scim.ScimUserProvisioning;
 import org.cloudfoundry.identity.uaa.scim.exception.ScimException;
 import org.cloudfoundry.identity.uaa.security.beans.SecurityContextAccessor;
-import org.cloudfoundry.identity.uaa.zone.MultitenancyFixture;
+import org.cloudfoundry.identity.uaa.zone.IdentityZone;
+import org.cloudfoundry.identity.uaa.zone.IdentityZoneHolder;
+import org.cloudfoundry.identity.uaa.zone.beans.IdentityZoneManager;
+import org.junit.After;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
-import org.mockito.Mockito;
+import org.mockito.MockedStatic;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.security.core.authority.AuthorityUtils;
 
-import java.util.Collection;
-import java.util.Collections;
-
-import static junit.framework.Assert.assertEquals;
-import static junit.framework.Assert.assertTrue;
-import static org.hamcrest.Matchers.containsString;
-import static org.hamcrest.Matchers.is;
-import static org.mockito.ArgumentMatchers.anyString;
-import static org.mockito.Mockito.when;
-
 /**
  * @author Dave Syer
  * @author Luke Taylor
- *
  */
 public class UserIdConversionEndpointsTests {
 
     @Rule
     public ExpectedException expected = ExpectedException.none();
 
-    private IdentityProviderProvisioning provisioning = Mockito.mock(IdentityProviderProvisioning.class);
-
     private UserIdConversionEndpoints endpoints;
 
     private SecurityContextAccessor mockSecurityContextAccessor;
 
-    private ScimUserEndpoints scimUserEndpoints = Mockito.mock(ScimUserEndpoints.class);
+    private final ScimUserEndpoints scimUserEndpoints = mock(ScimUserEndpoints.class);
+
+    private final ScimUserProvisioning scimUserProvisioning = mock(ScimUserProvisioning.class);
+
+    private final MockedStatic<IdentityZoneHolder> idzHolderMockedStatic;
+    private final IdentityZoneManager identityZoneManager = mock(IdentityZoneManager.class);
 
     @SuppressWarnings("rawtypes")
-    private Collection authorities = AuthorityUtils
-                    .commaSeparatedStringToAuthorityList("orgs.foo,uaa.user");
+    private final Collection authorities = AuthorityUtils
+            .commaSeparatedStringToAuthorityList("orgs.foo,uaa.user");
+
+    public UserIdConversionEndpointsTests() {
+        this.idzHolderMockedStatic = mockStatic(IdentityZoneHolder.class);
```

Comment:
Sad that we have to use mockStatic here. But I'll trust that this won't cause any additional flakiness in the test results.

Reply:
I was able to get rid of this mock entirely (commit 2644f53).